_posts/2018-03-23-cache-breaking-for-faster-web-application-updates.md

```diff
@@ -10,26 +10,28 @@ tags: []
+What if we could cache the majority of our application code to keep it loading fast, but also get updates to our users the first time they load our application after a deploy? The key to making this possible is to disable client-side caching on the main application file, `index.html`, and then aggressively cache all other resources. This keeps our application load time down since `index.html` is nothing more than a [50+ line shell around the application](https://github.com/agrc/deq-enviro/blob/c11865a477be1d5970c457636d9c738df58483e0/_src/index.html). 
+
+Here's an example of new headers for this file:
 
 ![screenshot of headers]({{ '/images/index_headers.png' | prepend: site.baseurl }}){: .flex .flex--center .outline }
 
-`Cache-Control: no-cache` directs the browser to always check the server for a new version of this file. The bulk of the code for our applications are built into the JS & CSS files. For these files I added a timestamp to the request (e.g. `/dojo/dojo.js?rel=1519311093828`). This timestamp, implemented via [grunt-cache-breaker](https://github.com/shakyShane/grunt-cache-breaker), is changed each time the application is built. Browsers cache resources by unique URL. So, when the URL to the resource is changed, the browser will instantly request a new version from the server. No more wondering if my users are getting updates!
+"`Cache-Control: no-cache`" directs the browser to always check the server for a new version of this file. The bulk of the code for our applications are built into the JS and CSS files. For these files, I added a time stamp to the request (e.g., `/dojo/dojo.js?rel=1519311093828`). This time stamp, implemented via [grunt-cache-breaker](https://github.com/shakyShane/grunt-cache-breaker), is changed each time the application is built. Browsers cache resources by unique URL. So, when the URL to the resource is changed, the browser will instantly request a new version from the server. No more wondering whether my users are getting updates!
 
-Bonus tip: The [`ETag`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) header allows the browser to check to see if a resource has changed on the server before actually downloading it (thus the `304` status code).
+_**Bonus tip**_: The [`ETag`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) header allows the browser to check whether a resource has changed on the server before actually downloading it (thus the `304` status code).
```

Comment:
* Regarding the whole sentence: I'm not completely sure how to read this sentence (and I'm sure it's partially because of my own ignorance! But maybe we can clarify a little bit?). 
  *  _"on the server"_: do we mean "The ETag header allows the browser to check whether a resource has changed on the server" or "The ETag header allows the browser to check whether a resource on the server has changed"?
  * _"before actually downloading it"_: What is doing the downloading? The browser?...And what is being downloaded? The resource? The ETag? The server?... 

(Again, I'm sure maybe this would make sense to someone who actually understands this stuff, but for readability, it may be worth spelling it out.)

Reply:
Etags work when a request for a resource is sent to a server with the etag header, the server can look at the tag and say, I have the same tag so I'm not going to waste time sending you the resource since you already have the same version as I do.